### ISSUE TITLE
Sync Options

### ISSUE THREAD

**Issue by Bamajon1974** [sienori/Tab-Session-Manager#521]
Does tab session manager extension (Chrome or Firefox) have any native session sync capability? I didn't notice any such features. Otherwise, does it sync using FireFox Sync or via Google Drive? 

Thanks in advance!

**Comment by sienori** [comment]
In version 6.0.0, I finally implemented the cloud sync function!!
You can save sessions to Google Drive and sync with other devices. #73
![cloudsync](https://user-images.githubusercontent.com/29514424/78505793-c027ba80-77b0-11ea-80ed-de25f8e67552.png)
![cloudsync2](https://user-images.githubusercontent.com/29514424/78505794-c322ab00-77b0-11ea-8347-2744ea0f6494.png)

**Comment by Bamajon1974** [comment]
Sweet! Thank you! I will try out the feature shortly.

**Comment by Baldomo** [comment]
Could any other syncing service (e.g. WebDAV/Nextcloud) be a reasonable effort to implement? Just asking because I don't personally maintain chrome extension with sync capabilities

**Comment by drkui** [comment]
+1 on WebDAV/Nextcloud support since I don't use Google Drive

**Comment by JonIcenhower** [comment]
+2 I'm piling on for this along with @drkui I don't use Google Drive either.
A Cloud agnostic solution would be the preferred method IMHO, thanks.

**Comment by CuriousCocainist** [comment]
+3 Google-less sync solution would be awesome!

**Comment by candrapersada** [comment]
is it possible to sync the settings "Tab Session Manager"

**Comment by nullpenguinexception** [comment]
It would be much better than GDrive if you can implement WebDAV.

**Comment by lucker999** [comment]
+1 for Nextcloud or any WebDAV solution!

**Comment by hollisticated-horse** [comment]
+1 To be able to sync across multiple browsers and machines using WebDav/Nextcloud or some sync link that does not need a google account :D

**Comment by claudio-salvio** [comment]
Maybe git or gist ...

I 💙 TSM

**Comment by cls-nebadje** [comment]
Yep, Nextcloud would be awesome! I just chipped in some support via PayPal. TSM is really great!

**Comment by totaltentigers** [comment]
Definitely would love other options besides Google Drive for sync. 

Even an option to automatically backup the session to a file, where I could then use a 3rd party service to sync that file.

**Comment by soredake** [comment]
https://wiki.mozilla.org/Services/Sync/Addon_Sync

**Comment by benjaminyzhang** [comment]
+1 for WebDAV solution!

**Comment by RunningDroid** [comment]
#951 is kind of related to this, Google returns 403 forbidden when trying to download a session containing URLs Google doesn't like.

I came across that bug after saving a session that has some IPFS URLs: https://gist.github.com/RunningDroid/0aebf9c2d55d5b27c5dfa14466baf2c5

(I saved it because IPFS-FPS can cause Firefox to crash & I wanted to look into it later)

**Comment by SmurfDude** [comment]
One Idea behind using the Firefox browser is to have Google not follow all your web movements. That's why I would gladly support having a sync option other than Google Drive.

**Comment by almereyda** [comment]
TSM being an addon for Firefox, it may seem preferable to try native synchronisation options available to the platform first, as suggested by @soredake.

Unfortunately, this comes at a price. Taken from https://wiki.mozilla.org/Services/Sync/Addon_Sync#Are_Add-on_Settings_Synchronized.3F:

> For add-on authors, to sync settings using Sync, there are effectively 3 methods.
>
> 1. Synchronize preferences (see https://developer.mozilla.org/en-US/docs/Archive/Mozilla/Firefox_Sync/Syncing_custom_preferences)
> 2. Build a Sync engine (if you need to sync more than just preferences). If you need this, please drop by the #sync channel in [IRC](https://wiki.mozilla.org/IRC) so you can chat with a Sync developer (there are some important details not currently captured formally in any docs).
> 3. Build it yourself, not using Sync.

For option 3., if Google Drive works, then [solidproject.org](https://solidproject.org/), [remotestorage.io](https://remotestorage.io/), Nextcloud and other web storage APIs, such as Git APIs from GitLab and GitHub, eventually also Gitea, and further generic CRUD-via-HTTP stores, like CouchDB or S3, seem very reachable.
 
> ![grafik](https://user-images.githubusercontent.com/1645308/219821369-ae5a9274-86fc-4d3c-8bf3-511189103596.png)
>
> Stackedit.io workspaces

Adding support for libre Drive alternatives could help liberate sensitive usage data from commercial enclosure.

* https://nextcloud.com/de/blog/using-webdav-fs-to-access-files-in-nextcloud/
  * https://github.com/perry-mitchell/webdav-fs
* https://github.com/tentwentyfour/nextcloud-link
